evm/src/cpu/kernel/asm/core/precompiles/main.asm

```diff
@@ -0,0 +1,133 @@
+%macro handle_precompiles
+    // stack: new_ctx, kexit_info, callgas, address, value, args_offset, args_size, ret_offset, ret_size
+    PUSH %%after
+    DUP5
+    // stack: address, %%after, new_ctx, kexit_info, gas, address, value, args_offset, args_size, ret_offset, ret_size
+    %jump(handle_precompiles)
+%%after:
+    // stack: new_ctx, kexit_info, callgas, address, value, args_offset, args_size, ret_offset, ret_size
+%endmacro
+
+global handle_precompiles:
+    // stack: addr, retdest
```

Comment:
Maybe we should include the full stack state here, or `...` to hint that there's more. I think it's fine to omit stack items that aren't relevant, but in my mind they're relevant/required here since the precompiles expect that stack state.